File: src/main/java/com/ultreon/ultranlang/exception/ErrorCode.java
package com.ultreon.ultranlang.exception;

public enum ErrorCode {
    UNEXPECTED_TOKEN("Unexpected token"),
    ID_NOT_FOUND("Identifier not found"),
    DUPLICATE_ID("Duplicate id found"),
    DEBUG("Debug test"),
    ;

    private final String value;

    ErrorCode(final String value) {
        this.value = value;
    }

    public String getValue() {
        return this.value;
    }
}

File: src/commonMain/kotlin/com/ultreon/ultranlang/error/ErrorCode.kt
package com.ultreon.ultranlang.error

enum class ErrorCode(val value: String) {
    UNEXPECTED_TOKEN("Unexpected token"),
    ID_NOT_FOUND("Identifier not found"),
    DUPLICATE_ID("Duplicate id found"),
    DEBUG("Debug test"),
    UNEXPECTED_STATEMENT_END("Unexpected statement end"),
}